rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -163,6 +174,26 @@ void SequentialWriter::close()
   storage_factory_.reset();
 }
 
+void SequentialWriter::take_snapshot()
+{
+  circular_message_cache_->swap_buffers();
```

Comment:
Should we check first that the output buffer is done being consumed? The behavior might be pretty strange if it gives back a partially-consumed buffer, where we will still have those older messages, that could then be written out alongside newer messages in a future snapshot.

I am thinking that if we have to prioritize a behavior, I think it would be "fully consume previous snapshot" before trying to do the new one.